src/server/pfs/cmds/cmds.go

```diff
@@ -442,27 +442,50 @@ $ {{alias}} foo@master --from XXX`,
 				toCommit = branch.NewCommit("")
 			}
 
+			if all && originStr != "" {
+				return errors.New("cannot specify both --all and --origin")
+			}
+
+			origin, err := parseOriginKind(originStr)
+			if err != nil {
+				return err
+			}
+
+			listClient, err := c.PfsAPIClient.ListCommit(c.Ctx(), &pfs.ListCommitRequest{
+				Repo:       branch.Repo,
+				From:       fromCommit,
+				To:         toCommit,
+				Number:     number,
+				All:        all,
+				OriginKind: origin,
+			})
+			if err != nil {
+				return grpcutil.ScrubGRPC(err)
+			}
+
 			if raw {
 				encoder := cmdutil.Encoder(output, os.Stdout)
-				return c.ListCommitF(branch.Repo, toCommit, fromCommit, number, false, func(ci *pfs.CommitInfo) error {
+				return clientsdk.ForEachCommit(listClient, func(ci *pfs.CommitInfo) error {
 					return encoder.EncodeProto(ci)
 				})
 			} else if output != "" {
 				return errors.New("cannot set --output (-o) without --raw")
 			}
 			writer := tabwriter.NewWriter(os.Stdout, pretty.CommitHeader)
-			if err := c.ListCommitF(branch.Repo, toCommit, fromCommit, number, false, func(ci *pfs.CommitInfo) error {
```

Comment:
have we talked about removing ListCommitF entirely? feels a little late to do that, I guess